src/map/clif.c

```diff
@@ -19778,6 +19778,9 @@ static void clif_parse_SearchStoreInfo(int fd, struct map_session_data *sd)
 
 	cardlist = RFIFOP(fd, sizeof(struct PACKET_CZ_SEARCH_STORE_INFO) + blocksize * item_count);
 
+	if (item_count < 1)
+		return; // Should never happen
```

Comment:
In my opinion this check should be placed some lines above, prior to assigning a value to `item_count`.

```c
	...

	if (packet_len < sizeof(struct PACKET_CZ_SEARCH_STORE_INFO))
	{// minimum packet length
		ShowError("clif_parse_SearchStoreInfo: Malformed packet (expected length=%u, length=%d, account_id=%d).\n", (uint32)sizeof(struct PACKET_CZ_SEARCH_STORE_INFO), packet_len, sd->bl.id);
		return;
	}

	if (p->itemsCount < 1) // Should never happen
		return;

	type       = p->searchType;
	max_price  = p->maxPrice;
	min_price  = p->minPrice;
	item_count = p->itemsCount;
	card_count = p->cardsCount;
	itemlist   = &p->items[0];

	...
```